### PR TITLE
chore: Add allocation source to StreamReader

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #.

## Rationale for this change

There is a memory leak reported by Java Arrow in #324. When looking at the error message for debugging:

```
Memory was leaked by query. Memory leaked: (49152)
Allocator(ROOT) 0/49152/180352/9223372036854775807 (res/actual/peak/limit)
```

I figured that it is hard to know where the allocation comes from. In this PR, I added the source of allocators.

The error message is updated to:

```
Memory was leaked by query. Memory leaked: (49152)                                                                                                                                                                                     
Allocator(StreamReader/shuffle reader) 0/49152/180352/9223372036854775807 (res/actual/peak/limit)                                                                                                                                      
```

### CODE REVIEW

**Comment by viirya** [comment]
cc @andygrove @sunchao @comphead

File: common/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometBlockStoreShuffleReader.scala [hunk 1]
```diff
@@ -108,7 +108,7 @@ class CometBlockStoreShuffleReader[K, C](
               // Closes previous read iterator.
               currentReadIterator.close()
             }
-            currentReadIterator = new ArrowReaderIterator(channel)
+            currentReadIterator = new ArrowReaderIterator(channel, "shuffle reader")
```
Comment (comphead) [inline common/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometBlockStoreShuffleReader.scala:111]: Perhaps we can use a fully qualified class name?

Comment (viirya) [reply common/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometBlockStoreShuffleReader.scala:111]: It sounds too verbose. I changed to simple name instead.

Comment (comphead) [inline common/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometBlockStoreShuffleReader.scala:111]: simple name here?

**Review by comphead — APPROVED** [review]
lgtm thanks @viirya

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request merged**